sourced/ml/transformers/basic.py

```diff
@@ -289,17 +289,20 @@ def __call__(self, rows: RDD) -> RDD:
         return rows.flatMap(self.deserialize_uast)
 
     def deserialize_uast(self, row: Row):
-        if not row[EngineConstants.Columns.Uast]:
-            return
-        row_dict = row.asDict()
-        row_dict[EngineConstants.Columns.Uast] = []
-        for i, uast in enumerate(row[EngineConstants.Columns.Uast]):
-            try:
-                row_dict[EngineConstants.Columns.Uast].append(self.parse_uast(uast))
-            except:  # noqa
-                self._log.error("\nBabelfish Error: Failed to parse uast for document %s for uast "
-                                "#%s" % (row[Uast2BagFeatures.Columns.document], i))
-        yield Row(**row_dict)
+        try:
+            if not row[EngineConstants.Columns.Uast]:
+                return
+            row_dict = row.asDict()
+            row_dict[EngineConstants.Columns.Uast] = []
+            for i, uast in enumerate(row[EngineConstants.Columns.Uast]):
+                try:
+                    row_dict[EngineConstants.Columns.Uast].append(self.parse_uast(uast))
+                except:  # noqa
+                    self._log.error("\nBabelfish Error: Failed to parse uast #%s for repository "
+                                    "%s" % (i, row[Uast2BagFeatures.Columns.repository_id]))
+            yield Row(**row_dict)
+        except ValueError:
+            self._log.error("The RDD provided does not include uasts.")
```

Comment:
`if not row[EngineConstants.Columns.Uast]:` must check exactly this. If it is not enough, change that lane please